### PR TITLE
Refactor FnAsClassEmitter

### PR DESCRIPTION
**Pull request by Chemaclass**
## 🔖 Changes

- Refactored the `FnAsClassEmitter` class
- Added unit tests for it

### CODE REVIEW

**Review by JesusValera — APPROVED** [review]
Super clear what this class does with the tests. 👍🏼👍🏼

File: src/php/Compiler/Analyzer/Ast/PhpVarNode.php [hunk 1]
```diff
@@ -47,6 +48,14 @@ final class PhpVarNode extends AbstractNode
 
     private string $name;
 
+    public static function withReturnContext(string $name, ?SourceLocation $sourceLocation = null): self
+    {
+        $returnNode = NodeEnvironment::empty()
```
Comment (jenshaase) [inline src/php/Compiler/Analyzer/Ast/PhpVarNode.php:53]: `$returnNode` should be called `$returnEnv` or just `$env`

**Review by jenshaase — CHANGES_REQUESTED** [review]
Only a small naming thing. Otherwise it looks very good!

**Review by jenshaase — APPROVED** [review]

**Comment by jenshaase** [comment]
Thanks! It is merged now!

**Pull request merged**